### PR TITLE
python310Packages.vapoursynth: add a withPlugins to extend it

### PR DESCRIPTION
**Pull request by cid-chan**
###### Description of changes

Right now, doing

```nix
pkgs.python310Packages.vapoursynth.override {
  vapoursynth = pkgs.vapoursynth.withPlugins [ pkgs.ffms ];
}
```

would fail with`error: attribute 'version' missing`.

In addition, it adds `pythonPackages.vapoursynth.withPlugins` that achieves the same result as manually overriding vapoursynth.

###### Things done

*Note to reviewers:
Be aware that on darwin vapoursynth.withPlugins is currently failing. In fact, this feature is currently causing vapoursynth to fail compiling at all on darwin. Fixing this specific issue is out of scope of this PR.*

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/vapoursynth/default.nix [hunk 1]
```diff
@@ -1,5 +1,4 @@
-{ vapoursynth, cython, buildPythonPackage, unittestCheckHook }:
-
+{ vapoursynth, cython, buildPythonPackage, unittestCheckHook, pythonPackages }:
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/vapoursynth/default.nix:1]: Please revert this formatting change

Comment (SuperSandro2000) [reply pkgs/development/python-modules/vapoursynth/default.nix:1]:
```suggestion
{ vapoursynth, cython, buildPythonPackage, unittestCheckHook, pythonPackages }:

```

File: pkgs/development/python-modules/vapoursynth/default.nix [hunk 2]
```diff
@@ -1,4 +1,4 @@
-{ vapoursynth, cython, buildPythonPackage, unittestCheckHook }:
+{ vapoursynth, cython, buildPythonPackage, unittestCheckHook, pythonPackages }:
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/vapoursynth/default.nix:1]:
```suggestion
{ vapoursynth, cython, buildPythonPackage, unittestCheckHook, python }:
```

to make sure that we use the same python version

File: pkgs/development/python-modules/vapoursynth/default.nix [hunk 3]
```diff
@@ -19,6 +19,13 @@ buildPythonPackage {
 
   unittestFlagsArray = [ "-s" "$src/test" "-p" "'*test.py'" ];
 
+  passthru = {
+    withPlugins = plugins:
+      pythonPackages.vapoursynth.override {
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/vapoursynth/default.nix:24]:
```suggestion
      python.pkgs.vapoursynth.override {
```

**Review by thiagokokada — APPROVED** [review]

**Pull request merged**